app/core/modals/Modal.tsx

```diff
@@ -0,0 +1,91 @@
+import { Dialog, Transition } from "@headlessui/react"
+import React, { Fragment } from "react"
+
+export const Modal = (props) => {
+  const {
+    isOpen,
+    setIsOpen,
+    title,
+    body,
+    onSubmit,
+    primaryAction = "Ok",
+    primaryButtonClass,
+    secondaryAction = "Cancel",
+    secondaryButtonClass,
+  } = props
```

Comment:
```suggestion
export const Modal = ({
    isOpen,
    setIsOpen,
    title,
    body,
    onSubmit,
    primaryAction = "Ok",
    primaryButtonClass,
    secondaryAction = "Cancel",
    secondaryButtonClass,
  }) => {
```

This is a bit confusing - if I now use a prop that isn't required the `Modal` doesn't throw (I think). I would bring the props into the component explicitly.

Reply:
Thanks Chris!

The suggested change throws a type error since the parameter's type is expecting any but not undefined.


```
Property 'secondaryButtonClass' is missing in type '{ title: string; body: Element; primaryAction: string; primaryButtonClass: string; isOpen: boolean; setIsOpen: Dispatch<SetStateAction<boolean>>; onSubmit: () => Promise<...>; }' but required in type '{ isOpen: any; setIsOpen: any; title: any; body: any; onSubmit: any; primaryAction?: string | undefined; primaryButtonClass: any; secondaryAction?: string | undefined; secondaryButtonClass: any; }'.ts(2741)
```

@chartgerink Do you think it's a good idea to develop type definitions for these parameters?

Reply:
Yes it would be good to do Type definitions so that the properties of the function are defined. I didn't test the suggestion so we can go two routes:
1. Fix the issue you raised and then request another review
2. Open an issue so that we bring the props into the component explicitly at a later, but soon, time. We keep as is for now.

Let me know which option you prefer. 😊

Reply:
Thanks for your ideas, Chris! 🙏 

I tried fixing the error by defining the prop types in this PR, using the `prop-types` approach. Would that work for you?